services/offline/src/lib/online-status.ts

```diff
@@ -50,5 +66,12 @@ export function useOnlineStatus(
         }
     }, [updateState])
 
-    return { online, offline: !online }
+    // Only fetch if `online === false` as local storage is synchronous and disk-based
+    const lastOnline = !online && localStorage.getItem(lastOnlineKey)
```

Comment:
minor: `useMemo` might help mitigate possible performance issues here - without it any time the parent rerenders it might create a new options object and cause an unnecessary localStorage lookup - so long as we can assume the value of localStorage.getItem(lastOnlineKey) will never change once `offline` is set.

```suggestion
    const lastOnline = !online && useMemo(localStorage.getItem(lastOnlineKey), [online])
```